File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let left = 0;
    let right = 0;
    let length = Infinity;
    let sum = 0;
    while(left <= nums.length){
        // console.log(`left is ${left} and right is ${right} and sum is ${sum}`)
        if(sum<target && right<nums.length){
            sum += nums[right];
            right++;
        } else {
            sum -= nums[left];
            left++;
        }
        if(sum >= target){
            length = Math.min(length, right - left);
        }
    }
    return length === Infinity ? 0 : length;
};

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l,r=0,-1
        total=0
        min_len=float('inf')
        
        while r<len(nums):
            if total<target:
                r+=1
                if r==len(nums): break
                total+=nums[r]
            else:
                l+=1
                total-=nums[l-1]
                
            if total>=target:
                min_len=min(min_len,r-l+1)
                
        return min_len if min_len != float('inf') else 0